020-Getting-started/070-nextjs.mdx

```diff
@@ -50,185 +46,621 @@ apiKey=YOUR_API_KEY_HERE
+from our Next.js application. To do this, run:
+
+```sh
 xata pull main
 ```
 
-## Create a new page to output our posts
+<details>
+  <summary><code>xata pull main</code> output</summary>
 
-Create a new file at `src/pages/posts.tsx` and start with a typical Hello World output. Next JS will render and pages added to the `pages` directory as distinct URL. This means we can find our new page at https://localhost:3000/posts
+```sh
+$ xata pull main
+Successfully pulled 1 migrations from main branch
+Running codegen...
+Generated Xata code to ./src/xata.ts
+```
 
-```tsx
-export default function PostsListsPage() {
+</details>
+
+This updates the contents of `src/xata.ts` based on the schema defined on the `main` branch of our database. So, if you
+make any further changes to the schema, run `xata pull <branch>` to update the auto-generated code.
+
+## Query and list the posts
+
+Now we're ready to integrate Xata into the Next.js codebase. Let's start by stripping back the landing page,
+`src/app/page.tsx`, to a bare template:
+
+```tsx title="src/app/page.tsx"
+export default async function Home() {
   return (
-    <>
-      <h1>My xata posts</h1>
-    </>
+    <main className="flex flex-wrap justify-center content-start p-8 lg:p-24 min-h-screen">
+      <div className="z-10 h-50 w-full max-w-5xl items-center justify-between text-xl lg:flex">
+        <p className="fixed left-0 top-0 flex w-full justify-center pb-6 pt-8 lg:static lg:w-auto bg-gradient-to-b from-white via-white via-65% dark:from-black dark:via-black lg:bg-none">
+          <a href="/">Get started with Xata and Next.js</a>
+        </p>
+        <div className="fixed bottom-0 left-0 flex w-full items-end justify-center bg-gradient-to-t from-white via-white dark:from-black dark:via-black lg:static lg:h-auto lg:w-auto lg:bg-none">
+          <a href="https://xata.io" className="w-20">
+            <img src="https://raw.githubusercontent.com/leggetter/xata-examples/5b0715c9ba36556400b6121bd224203cb89f7741/docs/app_logo.svg" />
+          </a>
+        </div>
+      </div>
+
+      <div className="w-full max-w-5xl mt-16"></div>
+    </main>
   );
 }
 ```
 
-Next let's import the `getXataClient` function which was provided to us by the `xata init` command that was run earlier. It should have placed the file in `src/xata.ts`. You'll notice that if we save this file, Next.js will error out with the following message.
+Next, import the auto-generated `getXataClient` function from `src/xata.ts`, get all the posts using the client, and
+list them within the page:
 
-<Alert status="warning">
-  Error: You are trying to use Xata from the browser, which is potentially a non-secure environment. If you understand
-  the security concerns, such as leaking your credentials, pass `enableBrowser: true` to the client options to remove
-  this error.
-</Alert>
+```tsx title="src/app/page.tsx" {1-3,6,21-41}
```

Comment:
I think it needs one more line to include the final `</div>`.

```suggestion
```tsx title="src/app/page.tsx" {1-3,6,21-42}
```